Docs 6314 csm update nov2 (#40)
mulesoft/docs-anypoint-security: modules/ROOT/pages/asm-secret-group-creation-task.adoc

* Start with a letter
* Contain only letters, numbers, and dashes (but cannot end with a dash)

After creating your secret group, it appears in the *Secret Groups* list view. You can then add the necessary secret types by adding a truststore or a keystore.

== Add a Truststore

Add a truststore for public certificates of trusted servers. The truststore stores up to 15 certificates from the trusted CA, which are then used to verify certificates presented by the server in an SSL connection.

. In the Secret Groups list view, select the secret group to receive a new truststore, and click *Edit*. 
. Select *Truststore* in the menu on the left, and click *Add Truststore*. 
. In the *Add Truststore* screen, add the required information. 
+
* *Name* +
Enter a name for your keystore.
* *Type* +
Select the truststore type from the drop-down menu. +
Supported types are:
+
** PEM
** JKS
** PKCS12
** JCEKS
* *Truststore File* +

> Click *Choose File*, and select the truststore file to upload. 

Reviewer: ```suggestion
Click *Choose File* and select the truststore file to upload. 
```

Specify the type of keystore to add to the secret group. The keystore is the combination of the authorization certificate, its corresponding private keys, and the certification authority's path.

. In the *Secret Groups* list view, select the secret group to which to add a keystore, and click *Edit*. 
. Select *Keystore* in the menu on the left, and click *Add Keystore*. 
. In the *Name* field, enter a name for your keystore.
. In the *Type* field, select the keystore type from the drop-down menu.
+
Supported types include:
+

> * Privacy Enhanced Mail (PEM) +

Reviewer: ```suggestion
* Privacy-Enhanced Mail (PEM) +
```

> Base64 encoded ASCII file with a `cer`, `crt`, or `pem` extension.

Reviewer: ```suggestion
Base64-encoded ASCII file with a `cer`, `crt`, or `pem` extension
```

* Java Keystore (JKS) +

> Repository for authorization or public key certificates. The JKS keystore type does not store secret keys.

Reviewer: ```suggestion
Repository for authorization or public key certificates
The JKS keystore type does not store secret keys.
```

> * PKCS12 +

Reviewer: ```suggestion
* PKCS #12 +
```

> Stores server and intermediate certificates in an archive file format. The PKCS12 keystore type does not store secret keys.

Reviewer: ```suggestion
Stores server and intermediate certificates in an archive file format
The PKCS #12 keystore type does not store secret keys.
```

> * Java Cryptography Extension KeyStore (JCEKS) +

Reviewer: ```suggestion
* Java Cryptography Extension keystore (JCEKS) +
```

> Stores server and intermediate certificates as well as secret keys.

Reviewer: ```suggestion
Stores server and intermediate certificates as well as secret keys
```

.. To add a *PEM* type keystore, you must provide: 
+
** *Certificate File* +
Click *Choose File* to locate and upload the PEM certificate file.  
** *Key File* +
Click *Choose File* to locate and upload the PEM formatted file that contains the private key for the certificate. 
** *Key Passphrase* +
Enter the word or phrase that protects the private key. 

> ** *CA Path Certificate File* +

Reviewer: CA = certificate authority or certificate-based authentication?

Author: certificate authority

Click *Choose File* to locate and upload the certificate signed by a certification authority (CA). +
The CA path contains the intermediary and root certificates that are related to the certificate file you want to use.    
** *Override Expiration Date* +
Select the date to override the default expiration date of the certificate.
.. To add a *JKS*, *PKCS12*, or *JCEKS* type keystore, you must provide: 
+
** *Keystore File* +
Click *Choose File* to locate and upload the keystore file to use.
** *Keystore Passphrase* +
Enter the word or phrase that protects the keystore.
** *Alias* +

> The alias used to access the keystore entries (key and trusted certificate entries).

Reviewer: Not parallel with peer definitions. Does the user enter this (Enter the...), or is this already specified (Specifies the...)?

** *Key Passphrase* +

> The word or phrase that protects the private key.

Reviewer: ditto re. parallelism

** *Algorithm* +

> The algorithm to use for encryption of keys.

Reviewer: ditto

** *Override Expiration Date* +
Select the date to override the current expiration date of the certificate.
. Click *Save*.

== Add a Certificate Pinset

Add a concatenated list of PEM certificates to the secret group. 

. In the *Secret Groups* list view, select the secret group to which to add a certificate pinset, and click *Edit*. 
. Select *Certificate Pinset* in the menu on the left, and click *Add Certificate Pinset*. 
. In the *Certificate Pinset* screen, add the required information. +
* *Name* +
Enter a name for the certificate pinset. 
* *Certificate File* +

> Click *Choose File* and select the PEM formatted CA certificate to upload.

Reviewer: ```suggestion
Click *Choose File* and select the PEM-formatted CA certificate to upload.
```
Is "CA certificate" redundant?

* *Expiration Date* +
Select the expiration date for the certificate.
. Click *Save*.

== Add a Shared Secret

Add a shared secret users can use for authentication. 

. In the *Secret Groups* list view, select the secret group to which to add a shared secret, and click *Edit*. 
. Select *Shared Secret* in the menu on the left, and click *Add Shared Secret*. 
. In the *Add Shared Secret* screen, add the required information. 
+
* *Name* +
Enter a name for your shared secret.
* *Type* +
Select the shared secret type from the drop-down menu.
 ** *Username Password* +
 Provide a username and password.
 ** *Symmetric Key* +
 Provide a Base64 string containing symmetric key.
 ** *S3 Credential* +
 Provide the access key ID and the secret access key to an S3 bucket.
 ** *Blob* +

>  Provide a base64-encoded value.

Reviewer: ```suggestion
 Provide a Base64-encoded value.
```

. Click *Save*. 

//// 
== Add a List of Revoked Certificates 

A Certificate Revocation List (CRL) distributor is an entity that creates and maintains a list of certificates issued by the CA that are no longer trusted because their associated private keys, or a signing CA, were compromised.

> Anypoint Secrets Manager lets you create CRL distribution points to query a CRL Distributor for a specific CA certificate. +

Reviewer: ```suggestion
The secrets manager enables you to create CRL distribution points to query a CRL distributor for a specific CA certificate. 
```
secrets manager is not an Anypoint Platform product

> To configure a CRL distributor, you must provide a URL of the CRL distributor, a CA certificate, and a frequency to check the list for updates. +

Reviewer: ```suggestion
To configure a CRL distributor, you must provide the URL of the CRL distributor, a CA certificate, and a frequency to check the list for updates. +
```

> As some CRL lists can grow very large, you can provide an optional Delta CRL Issuer URL (if the CRL distributor supports it) to retrieve deltas to the CRL list, instead of retrieving the entire list.

Reviewer: ```suggestion
Because some CRL lists can grow very large, you can provide an optional Delta CRL issuer URL (if the CRL distributor supports it) to retrieve deltas to the CRL list, instead of retrieving the entire list.
```

* *Delta CRL Issuer URL* (Optional) - Add the URL for the list of all certificates revoked since the last time a complete CRL was created. 
* *Expiration Date* (Optional) - Select the expiration date for the certificate.
. Click *Save*.
////

== Edit a Secret Group

When you edit a secret group, other users are locked out while you edit the necessary security objects. If a user doesn't end an editing session by either finishing or canceling it, the *Edit* button is disabled for other users. If you click *Cancel Edit*, the open editing session is closed, and the *Edit* button becomes available, even if the session was started by a different user. 

. In the *Secret Group* list view, click *Edit* for the secret group that you want to update. 
. Make your changes and click *Finish*. +

> The secret group is updated with the changes, and Secrets Manager serves the updated secrets to all redeployed applications after this point. 

Reviewer: ```suggestion
The secret group is updated with the changes, and the secrets manager serves the updated secrets to all redeployed applications after this point. 
```

. Click *Cancel Edit* to close an open editing session. +

> Use caution when you cancel another user's editing session, as all the changes made by the other user will be lost.

Reviewer: ```suggestion
Use caution when you cancel another user's editing session, because all the changes made by the other user will be lost.
```
